gensim/models/sent2vec.py

```diff
@@ -0,0 +1,307 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import division
+import logging
+import numpy as np
+from numpy import dot
+from gensim import matutils
+from random import randint
+import sys
+import random
+from gensim.utils import SaveLoad, tokenize
+import time
+
+logger = logging.getLogger(__name__)
+# TODO: add logger statements instead of print statements
+# TODO: add docstrings and tests
+
+
+class Entry():
+    def __init__(self, word=None, count=0, subwords=[]):
+        self.word = word
+        self.count = 0
+        self.subwords = subwords
+
+
+class Dictionary():
+    def __init__(self, t, bucket, minn, maxn, max_vocab_size=30000000, max_line_size=1024):
+        self.max_vocab_size = max_vocab_size
+        self.max_line_size = max_line_size
+        self.words = []
+        self.word2int = [-1] * max_vocab_size
+        self.pdiscard = []
+        self.ntokens = 0
+        self.size = 0
+        self.t = t
+        self.bucket = bucket
+        self.maxn = maxn
+        self.minn = minn
+
+    def hash_(self, word):
+        h = 2166136261
+        for i in range(len(word)):
+            h = h ^ ord(word[i])
+            h = h * 16777619
+        return h
+
+    def find(self, word):
+        h = self.hash_(word) % self.max_vocab_size
+        while self.word2int[h] != -1 and self.words[self.word2int[h]].word != word:
+            h = (h + 1) % self.max_vocab_size
+        return h
+
+    def add(self, word):
+        h = self.find(word)
+        self.ntokens += 1
+        if self.word2int[h] == -1:
+            e = Entry(word=word, count=1)
+            self.words.append(e)
+            self.word2int[h] = self.size
+            self.size += 1
+        else:
+            self.words[self.word2int[h]].count += 1
+
+    def read(self, sentences, min_count):
+        minThreshold = 1
+        for sentence in sentences:
+            for word in sentence:
+                self.add(word)
+                if self.ntokens % 1000000 == 0:
+                    print "Read " + str(self.ntokens / 1000000) + "M words"
+                if self.size > 0.75 * self.max_vocab_size:
+                    minThreshold += 1
+                    self.threshold(minThreshold)
+
+        self.threshold(min_count)
+        self.initTableDiscard()
```

Comment:
same as for variables (no camel case), here and everywhere.

Reply:
Done. Kindly verify in the current commit.